### PR TITLE
:bug: Restore Gopkg.toml

### PR DESCRIPTION
**Pull request by DirectXMan12**
This restores Gopkg.toml, and adds in some minimal testing to ensure
that Gopkg.toml is up to date.  Testing matches would be difficult
due to differences in how go modules and dep capture the version of
branch and non-version-tag-based dependencies.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
cc @mengqiy

File: test.sh [hunk 1]
```diff
@@ -135,3 +135,21 @@ golangci-lint run --disable-all \
 header_text "running go test"
 
 go test -mod=vendor ./pkg/... ./cmd/... -parallel 4
+
+# ensure that Gopkg.{toml,lock} are up-to-date
+header_test "ensuring that Gopkg.{toml,lock} are up to date..."
```
Comment (mengqiy) [inline test.sh]: typo here. should be `header_text`

**Comment by mengqiy** [comment]
travis can't find `dep`. Please fix.

**Pull request merged**